### PR TITLE
add the insert methode. and few more needed ones

### PR DESCRIPTION
**Pull request by reahaas**

### CODE REVIEW

File: WireBSTNode.java [hunk 1]
```diff
@@ -5,20 +5,20 @@
 // import java.lang.*;
 
 
-public class WireBSTNode extends WireBST {
+public class WireBSTNode {
 
     private int _studentNum;
-    private String _name;
+    private String _studentname;
```
Comment (Dor8) [inline WireBSTNode.java]: change to _studentName

File: WireBSTNode.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public void setParent( WireBSTNode parent)
     // end of getters and setters for all the pointers of the node
 
     // getters and setters values of the node: studentNum, name
-    public int getStudentNum()
+    public void getStudentNum()
```
Comment (Dor8) [inline WireBSTNode.java]: return to int again

File: WireBSTNode.java [hunk 3]
```diff
@@ -68,17 +68,64 @@ public void setStudentNum(int studentNum)
...
+    }
+
+
+    /**
+     * equals between two nodes means that they have the same StudentNum.
+     * @param node check for equality
+     * @return true if node is equal to this node.
+     */
+    public boolean equal(WireBSTNode node){
+        if (this.getStudentNum() == node.getStudentNum())
```
Comment (reahaas) [inline WireBSTNode.java]: rewrite the if to be in one line: if(condition: do ? else do)

File: WireBSTNode.java [hunk 4]
```diff
@@ -68,17 +68,64 @@ public void setStudentNum(int studentNum)
...
+        if (this.getRight().getParent().equal(this) )
+            return true;
+        else
+            return false;
+    }
+
+
+    /**
+     *
+     * @return true for nodes that have left son, and false for nodes that poit on thier predecessor in thier left son (or null).
```
Comment (reahaas) [inline WireBSTNode.java]: poit->point
thier->their

File: WireBST.java [hunk 5]
```diff
@@ -28,10 +29,72 @@ public WireBSTNode getHead(){
...
+    public void setHead(WireBSTNode newHead){
+        _head = newHead ;
+    }
+
+
+
+    /**
+     * insert of the node WireBST
+     * @param studentNum
+     * @param studentname
```
Comment (Dor8) [inline WireBST.java]: change to studentName

File: WireBSTNode.java (hunk 2 above)
Comment (reahaas) [reply WireBSTNode.java]: done

File: WireBSTNode.java (hunk 1 above)
Comment (reahaas) [reply WireBSTNode.java]: done

File: WireBSTNode.java (hunk 4 above)
Comment (reahaas) [reply WireBSTNode.java]: done

File: WireBSTNode.java (hunk 3 above)
Comment (reahaas) [reply WireBSTNode.java]: done

**Pull request merged**